fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -160,12 +164,22 @@ public List<Column> listPartitionColumns(String databaseName, String tableName,
     public List<PartitionInfo> getPartitions(Table table, List<String> partitionNames) {
         try (Connection connection = getConnection()) {
             List<Partition> partitions = schemaResolver.getPartitions(connection, table);
+            String minInt = IntLiteral.createMaxValue(Type.INT).getStringValue();
+            String minDate = DateLiteral.createMaxValue(Type.DATE).getStringValue();
+
             ImmutableList.Builder<PartitionInfo> list = ImmutableList.builder();
-            if (partitions.size() > 0) {
+            if (!partitions.isEmpty()) {
                 for (Partition partition : partitions) {
-                    if (partitionNames.contains(partition.getPartitionName())) {
+                    String partitionName = partition.getPartitionName();
+                    if (partitionNames.contains(partitionName)) {
                         list.add(partition);
                     }
+                    // Determine boundary value
+                    if (partitionName.equalsIgnoreCase(PartitionUtil.MYSQL_PARTITION_MAXVALUE)) {
+                        if (partitionNames.contains(minInt) || partitionNames.contains(minDate)) {
```

Comment:
why?

Reply:
When updating a materialized view, a situation occurs where the maximum value in the partition of the materialized view is either the maximum int or the maximum max, but the maximum value in the partition of the base jdbc table is MaxValue. This will cause the partition to be missed during the refresh action when comparing partitions, resulting in an error. Implement a logic here that when the partition of the base table contains MAXVALUE and the materialized view contains the maximum int or date, the materialized view partition will return that maximum partition.

Reply:
英文水平不行，贴一遍中文：在更新物化视图时，会发生一个情况，物化视图的分区中，最大值是最大的int或者最大的max，但是base jdbc表的分区最大值是MaxValue，这在刷新物化视图这个动作比较分区的时候，会漏掉分区，产生错误。此处实现一个逻辑，当base表的分区中包含MAXVALUE且物化视图包含最大int或者最大date时，物化视图的分区列表中会返回那个最大分区。